code/app/src/main/java/com/example/sigma_blue/context/GlobalContext.java

```diff
@@ -75,41 +78,67 @@ public GlobalContext() {
      * Toggle if an item is in the list of highlighted items or not.
      * @param item
      */
-    public void toggleHighlightItem(Item item) {
+
+    public void toggleInsertSelectedItem(Item item) {
+
         if (!this.highlightedItems.contains(item)){
             this.highlightedItems.add(item);
         } else {
             this.highlightedItems.remove(item);
         }
+
+        this.getItemList().getListAdapter().notifyDataSetChanged();
+
     }
 
     /**
      * Return list of highlighted items
-     * @return
+     * @return a List of the selected item
      */
-    public ArrayList<Item> getHighlightedItems() {
+    public ArrayList<Item> getSelectedItems() {
+
         return this.highlightedItems;
     }
 
     /**
      * Clear highlighted items
      */
-    public void resetHighlightedItems() {
+    public void resetSelectedItems() {
         this.highlightedItems.clear();
-        this.itemList.getAdapter().resetHighlightedItems();
+    }
+
+    /**
+     * Updates the item list through the view adapter.
+     */
+    public void notifyItemChanged() {
+        this.getItemList().getListAdapter().notifyDataSetChanged();
+    }
+
+    /**
+     * Method for doing a set difference of the items stored and the selected
+     * items.
+     */
+    public void confirmItemDelete() {
```

Comment:
I don't think confirmItemDelete is a good name. maybe deleteSelectedItems?

Reply:
I have changed the method name to a more appropriate one.